akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -491,3 +500,78 @@ private[akka] final case class MapAsyncUnordered[In, Out](parallelism: Int, f: I
     if (todo > 0) ctx.absorbTermination()
     else ctx.finish()
 }
+
+/**
+ * INTERNAL API
+ */
+private[akka] final case class Log[T](name: String, extract: T ⇒ Any, logAdapter: Option[LoggingAdapter]) extends PushStage[T, T] {
+
+  import Log._
+
+  private var log: LoggingAdapter = _
+  private var logLevels: OperationAttributes.LogLevels = _
+
+  @inline private def isEnabled(ctx: Context[T], l: LogLevels ⇒ LogLevel): Boolean = {
+    if (logLevels eq null) logLevels = ctx.attributes.logLevels.getOrElse(DefaultLogLevels)
+    l(logLevels).asInt != OffInt
+  }
+
+  override def onPush(elem: T, ctx: Context[T]): SyncDirective = {
+    if (isEnabled(ctx, _.onElement)) {
+      initLogger(ctx) // TODO if we have preStart this could be called in it, instead of in all callbacks
```

Comment:
okay, convinced: you have my leave to add `preStart` and `postStop` (and replace initAsyncInput)